packages/csvviewer/src/widget.ts

```diff
@@ -196,6 +198,26 @@ class CSVViewerFactory extends ABCWidgetFactory<CSVViewer, DocumentRegistry.IMod
  * The namespace for the module implementation details.
  */
 namespace Private {
+  /**
+   * Try to auto-detect the delimiter to use for this file.
+   *
+   * @param text - The text from the file.
+   *
+   * @returns A string to use as the delimiter.
+   */
+  export
+  function detectDelimiter(text: string): string {
```

Comment:
I think we might want to use a bit more sophisticated logic. Maybe count the number of times the delimiter appears in the first few lines and make sure it is the same number?